src/named_pixel_mapper.rs

```diff
@@ -0,0 +1,122 @@
+use std::{error::Error, str::FromStr};
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub enum NamedPixelMapperType {
+    Mirror(bool),
+    Rotate(usize),
+}
+
+impl FromStr for NamedPixelMapperType {
+    type Err = Box<dyn Error>;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let parts: Vec<&str> = s.split(':').collect();
```

Comment:
It looks like you always expect a `:` in the string. In that case `split_once` might be a better fit.

Reply:
Thanks. I updated the code using `split_once`.